code/controllers/subsystem/dbcore.dm

```diff
@@ -62,68 +51,66 @@ SUBSYSTEM_DEF(dbcore)
+	if(connection_result)
+		var/connection_status = connection_result["status"]
+		world.log << "Database Connection Status: [connection_status]"
+		if(connection_status == "ok")
+			return TRUE
+		else
+			message_admins("WARNING! Database connection failed! Status: [connection_status]")
+			log_sql("WARNING! Database connection failed! Status: [connection_status]")
+			return FALSE
 
-	connection = new /datum/BSQL_Connection(BSQL_CONNECTION_TYPE_MARIADB, CONFIG_GET(number/async_query_timeout), CONFIG_GET(number/blocking_query_timeout), CONFIG_GET(number/bsql_thread_limit))
-	var/error
-	if(QDELETED(connection))
-		connection = null
-		error = last_error
-	else
-		SSdbcore.last_error = null
-		connectOperation = connection.BeginConnect(address, port, user, pass, db)
-		if(SSdbcore.last_error)
-			CRASH(SSdbcore.last_error)
-		UNTIL(connectOperation.IsComplete())
-		error = connectOperation.GetError()
-	. = !error
-	if (!.)
-		last_error = error
-		log_sql("Connect() failed | [error]")
-		++failed_connections
-		QDEL_NULL(connection)
-		QDEL_NULL(connectOperation)
+/datum/controller/subsystem/dbcore/proc/IsConnected()
+	var/list/connection_results = json_decode(rustg_sql_connected())
+	switch(connection_results["status"])
+		if("online")
+			return TRUE
+		if("offline")
+			return FALSE
+		if("err")
+			var/error_data = connection_results["data"]
+			message_admins("SQL error occured in IsConnected! Error: [error_data]")
+			log_sql("SQL error occured in IsConnected! Error: [error_data]")
+			return FALSE
+
+#warn "REMOVE THIS SHIT BEFORE SHIPPING I SWEAR TO GOD"
+/datum/controller/subsystem/dbcore/proc/TestArbitraryDBCallDebugging(query, parameters = "{}")
+	world.log << "RUNNING QUERY [query]"
+	var/query_results = rustg_sql_query_blocking(query, parameters)
+	world.log << query_results
+	world.log << "QUERY OVER"
 
 /datum/controller/subsystem/dbcore/proc/CheckSchemaVersion()
 	if(CONFIG_GET(flag/sql_enabled))
 		if(Connect())
 			log_world("Database connection established.")
-			var/datum/DBQuery/query_db_version = NewQuery("SELECT major, minor FROM [format_table_name("schema_revision")] ORDER BY date DESC LIMIT 1")
+			var/datum/DBQuery/query_db_version = NewQuery("SELECT major, minor FROM [format_table_name("schema_revision")] ORDER BY date DESC LIMIT 1", "{}")
 			query_db_version.Execute()
-			if(query_db_version.NextRow())
-				db_major = text2num(query_db_version.item[1])
-				db_minor = text2num(query_db_version.item[2])
-				if(db_major != DB_MAJOR_VERSION || db_minor != DB_MINOR_VERSION)
-					schema_mismatch = 1 // flag admin message about mismatch
-					log_sql("Database schema ([db_major].[db_minor]) doesn't match the latest schema version ([DB_MAJOR_VERSION].[DB_MINOR_VERSION]), this may lead to undefined behaviour or errors")
-			else
-				schema_mismatch = 2 //flag admin message about no schema version
-				log_sql("Could not get schema version from database")
+			UNTIL(!query_db_version.in_progress)

```

Comment:
this one is not async on purpose I'm pretty sure

Reply:
I'm using it as my test query right now.

Reply:
I'm using it as my test query right now.

Reply:
It also was async, Execute() defaults to async with no args

Reply:
oh, okay then

Reply:
can you put the LOOP inside the execute command then please, not outside of it.